test/vec_notation.lean

```diff
@@ -0,0 +1,26 @@
+import Mathlib.Data.Fin.VecNotation
+
+/-! These tests are testing `PiFin.toExpr` and fail with
+`local attribute [-instance] PiFin.toExpr` -/
+
+open Lean
+open Lean.Meta
+open Qq
+
+#eval do
+  let x : Fin 0 → ℕ := ![]
+  let .true ← isDefEq (toExpr x) q((![] : Fin 0 → ℕ)) | failure
+
+#eval do
```

Comment:
Is there a better pattern than `#eval do` here?

Reply:
I would like to define `#assert` (which can run MetaM code), but haven't done so yet. AFAIK this is as good as it gets for now.

Reply:
By the way, `isDefEq` doesn't do typechecking, so it might be good to do typechecks for `toExpr` and `toTypeExpr`. See https://github.com/leanprover-community/mathlib4/blob/master/test/DeriveToExpr.lean